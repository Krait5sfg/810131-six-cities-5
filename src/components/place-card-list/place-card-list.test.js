import React from 'react';
import renderer from 'react-test-renderer';
import PlaceCardList from './place-card-list';

const offers = [
  {
    city: ``,
    accommodation: {
      bedroomsCount: 1,
      features: [`Dishwasher`, `Towels`, `Washer`, `Fridge`, `Breakfast`, `Laptop friendly workspace`, `Baby seat`, `Air conditioning`],
      guestsLimit: 2,
      isPremium: true,
      price: 241,
      rating: 4.4,
      title: `The Joshua Tree House`,
      type: `room`,
    },
    cityLocation: {latitude: 51.225402, longitude: 6.776314, zoom: 13},
    coordinates: [51.237402, 6.797314],
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy `,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      isPro: true,
      name: `Angelina`,
    },
    id: 1,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`],
    isFavorite: false,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
  },
  {
    city: ``,
    accommodation: {
      bedroomsCount: 2,
      features: [`Laptop friendly workspace`, `Baby seat`, `Air conditioning`],
      guestsLimit: 4,
      isPremium: false,
      price: 300,
      rating: 4.5,
      title: `The Joshua Tree House`,
      type: `hotel`,
    },
    cityLocation: {latitude: 51.225402, longitude: 6.776314, zoom: 13},
    coordinates: [51.247402, 6.777314],
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you `,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      isPro: true,
      name: `Angelina`,
    },
    id: 2,
    images: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`],
    isFavorite: false,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
  },
];
// потому что в компоненте PlaceCardList есть другой компонент PlaceCard
jest.mock(`../place-card/place-card`, () => `PlaceCard'`);

it(`PlaceCardList render correctly`, () => {
  const tree = renderer.create(<PlaceCardList
    offers={offers}
    typePage={`main`}
    onFavoriteButtonClick={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
