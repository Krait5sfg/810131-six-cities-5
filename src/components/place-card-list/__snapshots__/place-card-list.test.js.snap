// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCardList render correctly on MainPage 1`] = `
<div
  className="cities__places-list tabs__content places__list"
>
  <PlaceCard'
    offer={
      Object {
        "accommodation": Object {
          "bedroomsCount": 0,
          "features": Array [],
          "guestsLimit": 0,
          "isPremium": true,
          "price": 0,
          "rating": 0,
          "title": "",
          "type": "",
        },
        "city": "",
        "cityLocation": Object {
          "latitude": 0,
          "longitude": 0,
          "zoom": 0,
        },
        "coordinates": Array [
          0,
          0,
        ],
        "description": "",
        "host": Object {
          "avatar": "",
          "isPro": true,
          "name": "",
        },
        "id": 1,
        "images": Array [],
        "isFavorite": true,
        "previewImage": "",
      }
    }
    onFavoriteButtonClick={[Function]}
    typePage="main"
  />
  <PlaceCard'
    offer={
      Object {
        "accommodation": Object {
          "bedroomsCount": 0,
          "features": Array [],
          "guestsLimit": 0,
          "isPremium": true,
          "price": 0,
          "rating": 0,
          "title": "",
          "type": "",
        },
        "city": "",
        "cityLocation": Object {
          "latitude": 0,
          "longitude": 0,
          "zoom": 0,
        },
        "coordinates": Array [
          0,
          0,
        ],
        "description": "",
        "host": Object {
          "avatar": "",
          "isPro": true,
          "name": "",
        },
        "id": 2,
        "images": Array [],
        "isFavorite": true,
        "previewImage": "",
      }
    }
    onFavoriteButtonClick={[Function]}
    typePage="main"
  />
</div>
`;

exports[`PlaceCardList render correctly on OfferPage 1`] = `
<div
  className="near-places__list places__list"
>
  <PlaceCard'
    offer={
      Object {
        "accommodation": Object {
          "bedroomsCount": 0,
          "features": Array [],
          "guestsLimit": 0,
          "isPremium": true,
          "price": 0,
          "rating": 0,
          "title": "",
          "type": "",
        },
        "city": "",
        "cityLocation": Object {
          "latitude": 0,
          "longitude": 0,
          "zoom": 0,
        },
        "coordinates": Array [
          0,
          0,
        ],
        "description": "",
        "host": Object {
          "avatar": "",
          "isPro": true,
          "name": "",
        },
        "id": 1,
        "images": Array [],
        "isFavorite": true,
        "previewImage": "",
      }
    }
    onFavoriteButtonClick={[Function]}
    typePage="offer"
  />
  <PlaceCard'
    offer={
      Object {
        "accommodation": Object {
          "bedroomsCount": 0,
          "features": Array [],
          "guestsLimit": 0,
          "isPremium": true,
          "price": 0,
          "rating": 0,
          "title": "",
          "type": "",
        },
        "city": "",
        "cityLocation": Object {
          "latitude": 0,
          "longitude": 0,
          "zoom": 0,
        },
        "coordinates": Array [
          0,
          0,
        ],
        "description": "",
        "host": Object {
          "avatar": "",
          "isPro": true,
          "name": "",
        },
        "id": 2,
        "images": Array [],
        "isFavorite": true,
        "previewImage": "",
      }
    }
    onFavoriteButtonClick={[Function]}
    typePage="offer"
  />
</div>
`;
